pkg/vectorize/exp/exp.go

```diff
@@ -0,0 +1,172 @@
+// Copyright 2022 Matrix Origin
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package exp
+
+import (
+	"math"
+
+	"github.com/matrixorigin/matrixone/pkg/container/nulls"
+)
+
+var (
+	expUint8   func([]uint8, []float64) ExpResult
+	expUint16  func([]uint16, []float64) ExpResult
+	expUint32  func([]uint32, []float64) ExpResult
+	expUint64  func([]uint64, []float64) ExpResult
+	expInt8    func([]int8, []float64) ExpResult
+	expInt16   func([]int16, []float64) ExpResult
+	expInt32   func([]int32, []float64) ExpResult
+	expInt64   func([]int64, []float64) ExpResult
+	expFloat32 func([]float32, []float64) ExpResult
+	expFloat64 func([]float64, []float64) ExpResult
+)
+
+func init() {
+	expUint8 = expUint8Pure
+	expUint16 = expUint16Pure
+	expUint32 = expUint32Pure
+	expUint64 = expUint64Pure
+	expInt8 = expInt8Pure
+	expInt16 = expInt16Pure
+	expInt32 = expInt32Pure
+	expInt64 = expInt64Pure
+	expFloat32 = expFloat32Pure
+	expFloat64 = expFloat64Pure
+}
+
+type ExpResult struct {
+	Result []float64
+	Nsp    *nulls.Nulls
```

Comment:
unlike the log function, the Exp function has the whole real number set **R** as its domain, I think it is redundant to return the NULL values, perhaps return only the []float64 result would suffice.

Reply:
Yes, you're right